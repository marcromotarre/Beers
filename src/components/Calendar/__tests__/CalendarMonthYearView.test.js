import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import CalendarMonthYearView from '../CalendarMonthYearView';

/** this component should:
 *  paint year and month info inside calendar
 *      - month its transformed in text with the month name (capital letters)
 *
 *  props:
 *      - month: number of the month
 *      - year: number of the year
 *
 */

describe('CalendarMonthYearView Component Snapshots', () => {
  it('CalendarMonthYearView Snapshots', () => {
    expect(
      renderer.create(<CalendarMonthYearView month={9} year={2017} />)
    ).toMatchSnapshot();

    expect(
      renderer.create(<CalendarMonthYearView month={9} />)
    ).toMatchSnapshot();

    expect(
      renderer.create(<CalendarMonthYearView year={2017} />)
    ).toMatchSnapshot();
  });
});

describe('CalendarMonthYearView width month and year', () => {
  it('CalendarMonthYearView Component should render all information', () => {
    const { container, getByTestId } = render(
      <CalendarMonthYearView month={9} year={2017} />
    );
    const calendar = container.children[0];
    expect(calendar.nodeName).toBe('SECTION');

    const calendarMonthYearViewContainer = getByTestId(
      'calendar-month-year-view-container'
    );
    expect(calendarMonthYearViewContainer.children.length).toBe(2);
    const monthLabel = calendarMonthYearViewContainer.children[0];
    expect(monthLabel.nodeName).toBe('LABEL');
    expect(monthLabel.innerHTML).toBe('SEPTEMBER');
    const yearLabel = calendarMonthYearViewContainer.children[1];
    expect(yearLabel.nodeName).toBe('LABEL');
    expect(yearLabel.innerHTML).toBe('2017');
  });
});

describe('CalendarMonthYearView only month', () => {
  it('CalendarMonthYearView Component should render all information', () => {
    const { debug, getByTestId } = render(<CalendarMonthYearView month={9} />);
    const calendarMonthYearViewContainer = getByTestId(
      'calendar-month-year-view-container'
    );
    expect(calendarMonthYearViewContainer.children.length).toBe(1);
    const monthLabel = calendarMonthYearViewContainer.children[0];
    expect(monthLabel.nodeName).toBe('LABEL');
    expect(monthLabel.innerHTML).toBe('SEPTEMBER');
  });
});

describe('CalendarMonthYearView only year', () => {
  it('CalendarMonthYearView Component should render all information', () => {
    const { debug, getByTestId } = render(
      <CalendarMonthYearView year={2017} />
    );
    const calendarMonthYearViewContainer = getByTestId(
      'calendar-month-year-view-container'
    );
    expect(calendarMonthYearViewContainer.children.length).toBe(1);
    const yearLabel = calendarMonthYearViewContainer.children[0];
    expect(yearLabel.nodeName).toBe('LABEL');
    expect(yearLabel.innerHTML).toBe('2017');
  });
});
