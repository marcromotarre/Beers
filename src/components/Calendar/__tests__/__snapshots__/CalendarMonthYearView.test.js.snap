// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarMonthYearView Component Snapshots CalendarMonthYearView Snapshots 1`] = `
<section
  className="sc-AxjAm eKZXqA"
>
  <div
    className="sc-AxiKw euRTWC"
    height="68.4931506849315px"
    width="100px"
  >
    <div
      className="sc-AxhCb hmAOOJ"
      data-testid="calendar-month-year-view-container"
    >
      <label
        className="sc-AxhUy jgGAZg"
      >
        SEPTEMBER
      </label>
      <label
        className="sc-AxgMl cxvnSJ"
      >
        2017
      </label>
    </div>
  </div>
  <svg
    className="sc-AxirZ ktnnZK"
    width={100}
  />
</section>
`;

exports[`CalendarMonthYearView Component Snapshots CalendarMonthYearView Snapshots 2`] = `
<section
  className="sc-AxjAm eKZXqA"
>
  <div
    className="sc-AxiKw euRTWC"
    height="68.4931506849315px"
    width="100px"
  >
    <div
      className="sc-AxhCb hmAOOJ"
      data-testid="calendar-month-year-view-container"
    >
      <label
        className="sc-AxhUy jgGAZg"
      >
        SEPTEMBER
      </label>
    </div>
  </div>
  <svg
    className="sc-AxirZ ktnnZK"
    width={100}
  />
</section>
`;

exports[`CalendarMonthYearView Component Snapshots CalendarMonthYearView Snapshots 3`] = `
<section
  className="sc-AxjAm eKZXqA"
>
  <div
    className="sc-AxiKw euRTWC"
    height="68.4931506849315px"
    width="100px"
  >
    <div
      className="sc-AxhCb hmAOOJ"
      data-testid="calendar-month-year-view-container"
    >
      <label
        className="sc-AxgMl cxvnSJ"
      >
        2017
      </label>
    </div>
  </div>
  <svg
    className="sc-AxirZ ktnnZK"
    width={100}
  />
</section>
`;
